specification/threatmodel.adoc

```diff
@@ -1,187 +1,296 @@
+service by TVMs must be prevented. At the same time, denial of service by
 host software is considered out of scope.
 
 [[design_survey]]
-=== TVM Security Requirements to address threat model 
+=== TVM Security Requirements to address threat model
 
 |===
-| Category  |  Security Criteria  |  CoVE Requirement |  Example methods to meet requirement | Description/Example | RVI HC/SIG/TG owner 
+| *Category*  |  *Security Criteria*  |  *CoVE Requirement* |  *Example methods
+ to meet requirement* | *Description / Example* | *RVI normative or
+non-normative Reference*
 
-| Memory Footprint | Stolen/reserved memory | Implementation-specific | Minimize reserved memory | Recording meta data of secure memory | AP-TEE TG to specify
+| Memory Assignment | Ability to make memory confidential or non-confidential |
+Required | MMU, MPU, MTT | Confidential memory should be dynamically allocated/
+unallocated as required | RISC-V Priv. ISA, Smmtt
 
-| Memory Assignment | Ability to make memory confidential or non-confidential | Required | MMU, MPU, PMA/MTT extension | Confidential memory should be dynamically allocated/unallocated as required | "AP-TEE to specify priv. architecture" 
+| TEE CPU State Protection | State Isolation | Required | Priv. levels (M,S,HS,
+U) and Execution context (ASID, VMID, SDID) | Prevent untrusted code from
+arbitrarily accessing/modifying TEE CPU state | Priv ISA, Sv, Smmtt
 
-| TEE CPU State Protection | State Isolation | Required | Confidential qualifier and privilege levels M S HS U | Prevent untrusted code from arbitrarily accessing/modifying TEE CPU state | AP-TEE TG to specify 
+| Memory Confidentiality | Memory isolation (read) | Required | cryptography
+ and/or MMU, MPU, MTT extension | Prevent untrusted components from reading
+TEE memory | Sv, Smmtt
 
-| Memory Confidentiality | Memory isolation (read)     | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent untrusted components from reading TEE memory       | AP-TEE TG specify 
-| Memory Confidentiality | Cipher text read prevention | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent untrusted code from accessing encrypted TEE memory | AP-TEE TG specify 
-| Memory Confidentiality | Per TEE encryption          | Implementation-specific | cryptography and/or MMU, MPU, PMA/MTT extension | Each VM has one or more unique keys                        | AP-TEE TG to recommend 
-| Memory Confidentiality | Memory encryption strength  | Implementation-specific | cryptography                                    | Encryption algorithm and key strength                      | AP-TEE TG to recommend 
-| Memory Confidentiality | Number of encryption keys   | Implementation-specific | cryptography                                    | Number of TEE keys supported                               | AP-TEE TG to recommend 
+| Memory Confidentiality | Cipher text read prevention | Required | cryptography
```

Comment:
> Cipher text read prevention 

I agree "prevent unauthorized access to memory assigned to a different domain/TVM, regardless of memory protection" is a sensible isolation guarantee.

But in a memory protection context it's about freshness guarantees. 

Maybe rephrase in terms of freshness requirements? Boot freshness; domain freshness; TVM freshness; temporal freshness; etc.?

And then do the same for replay protection, integrity protection.

And build a simple threat/mitigation table around that. 

Actual implementations will have different threat models for what is in/out of scope - software attacks, physical attacks, whatever - and what performance/security trade-offs they accepts.